lib/tpm2_session.c

```diff
@@ -86,6 +86,10 @@ void tpm2_session_set_nonce_caller(tpm2_session_data *data, TPM2B_NONCE *nonce)
     data->nonce_caller = *nonce;
 }
 
+void tpm2_session_get_nonce_tpm(tpm2_session *session, TPM2B_NONCE *nonce_tpm) {
+    *nonce_tpm = session->internal.nonceNewer;
```

Comment:
this is pass by copy, is this what you want?
```c
return session->internal.nonceNewer;
```
I would do:
```c
const TPM2B_NONCE * tpm2_get_nonce_tpm(tpm2_session *session) {
    return (const TPM2B_NONCE *) &session->internal.nonceNewer;
}
```

Looks like you just bring it back for  a `memcpy()` when generating the HMAC, so this avoids a copy.

Reply:
I don't think you even need that `const` cast, my local testing shows no compiler complaints.

Reply:
Implemented